social_media/accounts/views.py

```diff
@@ -0,0 +1,117 @@
+from django.contrib.auth import authenticate
+from django.shortcuts import render
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.authtoken.models import Token
+from .serializer import AccountRegistrationSerializer,AccountSerializer
+from .models import User
+
+
+# Create your views here.
+class UserDetailsView(APIView):
+
+    def get(self,request,username):
+        try:
```

Comment:
Dont use .get() inside a try-catch. 

This is a beginner mistake. 

Either use .filter() or .exist() followed by .get() 

Reason: Why use a method which can lead to an exception when we can anticipate/check for the user